main.py

```diff
@@ -1773,10 +1775,28 @@ def start_client(window: sublime.Window, config: ClientConfig):
         window.status_message("Starting " + config.name + "...")
     debug("starting in", project_path)
 
+    # Create a dictionary of Sublime Text variables
     variables = window.extract_variables()
-    expanded_args = list(sublime.expand_variables(os.path.expanduser(arg), variables) for arg in config.binary_args)
+    variables["sublime_path"] = os.path.dirname(sublime.executable_path())
+    variables["sublime_libs"] = ';'.join(sys.path)
 
-    client = start_server(expanded_args, project_path)
+    # Expand language server command line environment variables
+    expanded_args = list(
+        sublime.expand_variables(os.path.expanduser(arg), variables)
+        for arg in config.binary_args
+    )
+
+    # Merge/expand OS environment variables
+    environ = os.environ.copy()
+    for var, value in config.environ.items():
+        # Merge vars (e.g.: PATH=$PATH;mypath)
+        # For ease of editing, allow lists for each environment variable
+        if isinstance(value, list):
+            value = ';'.join(value)
```

Comment:
As this is the first implementation, let's avoid adding extra features from the start.

The sublime build system keeps it simple:
https://github.com/twolfson/sublime-files/blob/7d2589861ca00294734379943e8b7cffb7fbac88/Packages/Default/exec.py#L55-L58

Please remove the `list` support and the sublime expand_variables.

Reply:
> The sublime build system keeps it simple

It is one line extra compared to the linked solution and is an optional feature which does not need to be used by anyone and not really more complicated.

If people want, they can still add 2048 or more chars long and over 15 lines wrapped lists of paths in one single string. But if a bit more overview is desired, they can split it into a list of strings. As JSON does not support multiline strings it is the only way to something like that and its harmless.

I thought about removing the ';' deliminator as it may not work with Linux/OSX. This would turn the list in a real mulitline-string-only-replacement.

I used this strategy for GitGuter's status_message templete already, which is read and joined everytime something changes - not only upon startup.

https://github.com/jisaacks/GitGutter/blob/24e254e24ef4ca57586e6f5e750da15cb45d3a0d/modules/show_diff.py#L133

An it is included in the 0.2ms required to handle the environment variable expansion. Removing it will not save any time at all.